src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IFieldReferenceExpression.cs

```diff
@@ -245,6 +245,84 @@ void M()
             VerifyOperationTreeAndDiagnosticsForTest<MemberAccessExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
         }
 
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IFieldReferenceExpression_IndexingFixed_Unmovable()
+        {
+            string source = @"
+
+unsafe struct S1
+{
+    public fixed int field[10];
+}
+
+unsafe class C
+{
+    void M()
+    {
+        S1 s = default;
+
+        /*<bind>*/ s.field[3] = 1; /*</bind>*/
+    }
+}
+";
+            string expectedOperationTree = @"
+ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Int32) (Syntax: 's.field[3] = 1')
+  Left: 
+    IOperation:  (OperationKind.None, Type: null) (Syntax: 's.field[3]')
+      Children(2):
+          IFieldReferenceOperation: System.Int32* S1.field (OperationKind.FieldReference, Type: System.Int32*) (Syntax: 's.field')
+            Instance Receiver: 
+              ILocalReferenceOperation: s (OperationKind.LocalReference, Type: S1) (Syntax: 's')
+          ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 3) (Syntax: '3')
+  Right: 
+    ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 1) (Syntax: '1')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<AssignmentExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics,compilationOptions: TestOptions.UnsafeDebugDll);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IFieldReferenceExpression_IndexingFixed_Movable()
+        {
+            string source = @"
+
+unsafe struct S1
+{
+    public fixed int field[10];
+}
+
+unsafe class C
+{
+    S1 s = default;
+
+    void M()
+    {
+        /*<bind>*/ s.field[3] = 1; /*</bind>*/
+    }
+}
+";
+            string expectedOperationTree = @"
+ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Int32) (Syntax: 's.field[3] = 1')
+  Left: 
+    IOperation:  (OperationKind.None, Type: null) (Syntax: 's.field[3]')
```

Comment:
I'm surprised that the indexing operation is represented as `None`. I would have thought that IOperation supports indexing operations.

Reply:
They see "PointerIndexingExpressions" and it is "None".  I guess it is NYI yet.

---
In reply to: [176575249](https://github.com/dotnet/roslyn/pull/25660#discussion_r176575249) [](ancestors = 176575249)

Reply:
Thanks for the clarification. Chatted with @AlekseyTs and I think it's ok to leave as-is (IOperation supporting `PointerIndexingExpressions` is a pre-existing issue) given that the children are listed properly.